websites/I/iQIYI/presence.ts

```diff
@@ -56,144 +54,86 @@ presence.on("UpdateData", async () => {
-          .querySelector("h1")
-          .textContent.replace(document.querySelector("h1 a").textContent, "")
+  if (document.location.pathname.includes('/play') || document.location.pathname.includes("/intl-common/")){
+
+      const data = {
+          title: (document.querySelector('h1 a') || document.querySelector('title'))?.textContent,
+          ep: (document.querySelector('h1') || document.querySelector(".topice-source-list-item.item-active"))?.textContent.replace(document.querySelector('h1 a')?.textContent || "", '')
       },
-      URLItem: string = JSON["parse"](
-        document.querySelectorAll('script[type="application/ld+json"]')[1]
-          .innerHTML
-      )[0].itemListElement[0].item,
-      video: HTMLVideoElement = document.querySelector("video"),
-      timestamps: number[] = presence.getTimestampsfromMedia(video),
-      isMovie = URLItem.includes("movie"),
-      isVShow = URLItem.includes("variety-show"),
-      isTrial =
-        document.querySelector(
-          ".iqp-player-g.iqp-player .iqp-tip-stream .iqp-txt-vip"
-        )?.textContent !== undefined,
-      lastestEp: string[] = document
-        .querySelector("div.broken-line")
-        .nextSibling.nextSibling.nextSibling?.textContent.match(
-          /[1-9]?[0-9]?[0-9]/g
-        ),
-      contentEp: string[] = data.ep.match(/[1-9]?[0-9]?[0-9]/g),
-      isPreview =
-        lastestEp && contentEp
-          ? parseInt(contentEp[0]) > parseInt(lastestEp[0])
-          : false;
-
-    if (!data.ep && !isVShow && isMovie) data.ep = "Movie";
-    if (isVShow) data.ep = "Variety show";
-    if (!isVShow && !isMovie)
-      data.ep = `${(await strings).episode} ${contentEp[0]}`;
-
-    if (isTrial && !isPreview) data.ep = `${data.ep} (Trial)`;
-
-    if (video && !isNaN(video.duration)) {
-      if (isPreview && !isMovie && !isVShow) data.ep = `${data.ep} preview`;
-      else if (video.duration < 270 && !isMovie && !isPreview && !isTrial)
-        data.ep = "Highlight";
-
-      presenceData.details = data.title;
-      presenceData.state = data.ep;
-
-      presenceData.smallImageKey = video.paused ? "pause" : "play";
-      presenceData.smallImageText = video.paused
-        ? (await strings).pause
-        : (await strings).play;
-
-      presenceData.startTimestamp = timestamps[0];
-      presenceData.endTimestamp = timestamps[1];
-
-      if (showButtons) {
-        presenceData.buttons = [
-          {
-            label: isVShow
-              ? (await strings).watchVideo
-              : isMovie
-              ? (await strings).watchMovie
-              : (await strings).watchEpisode,
-            url: `https://www.iq.com/play/${
-              document.URL.split("?")[0].split("/")[4]
-            }`
+
+      URLItem: string = JSON['parse'](document.querySelectorAll('script[type="application/ld+json"]')[1]?.innerHTML || "{}")[0]?.itemListElement[0]?.item ?? document.URL,
+      video: HTMLVideoElement = document.querySelector('video'),
```

Comment:
why not JSON.parse() ?

Reply:
I will check If I can use it since the last time it returns error that I can't use JSON.parse()